unbound.go

```diff
@@ -244,12 +245,23 @@ func (u *Unbound) Resolve(name string, rrtype, rrclass uint16) (*Result, error)
 		r.Rr = make([]dns.RR, 0)
 		b := C.GoBytes(unsafe.Pointer(C.array_elem_char(res.data, C.int(j))), C.array_elem_int(res.len, C.int(j)))
 		for len(b) != 0 {
-			// Create the RR
+			// Create the RR; write out the header details and
+            // the rdata to a buffer, and unpack it again into an
+            // actual RR
 			h.Rdlength = uint16(len(b))
-			msg := make([]byte, 20+len(h.Name)) // Long enough
-			off, _ := dns.PackStruct(&h, msg, 0)
-			msg = msg[:off]
+
+			msg := make([]byte, len(h.Name) + 11)
+			off, _ := dns.PackDomainName(h.Name, msg, 0, nil, false)
+			binary.BigEndian.PutUint16(msg[off:], h.Rrtype)
+			off += 2
+			binary.BigEndian.PutUint16(msg[off:], h.Class)
+			off += 2
+			binary.BigEndian.PutUint32(msg[off:], h.Ttl)
+			off += 4
+			binary.BigEndian.PutUint16(msg[off:], h.Rdlength)
+			off += 2
 			rrbuf := append(msg, b...)
+
```

Comment:
Just exporting the pack/unpack for the header would fix this. Pondering a better fix. IIRC `res.data` holds the RR data without the header.